packages/browser/src/Component/Widget/Server/SPARQL/View/SparqlQueryWindow.js

```diff
@@ -146,45 +129,81 @@ export class SparqlQueryWindow extends EventSender {
         )
     );
 
-    this.addEventListener(Graph.EVENT_NODE_CLICKED, (node_text) => {
-      this.cityObjectProvider.selectCityObjectByBatchTable(
-        'gml_id',
-        URI.tokenizeURI(node_text).id
-      );
-      const cityObject = this.layerManager.pickCityObjectByBatchTable(
-        'gml_id',
-        URI.tokenizeURI(node_text).id
-      );
-      if (cityObject) {
-        focusCameraOn(
-          this.layerManager.view,
-          this.layerManager.view.controls,
-          cityObject.centroid,
-          {
-            verticalDistance: 200,
-            horizontalDistance: 200,
+    const fetchC3DTileFeatureWithNodeText = (node_text) => {
```

Comment:
could be a method in itownsutils to find a feature in a 3DTiles using a feature name (here 'gml_id') and a value (here node_text)

Reply:
note that it is possible to find multiple matching feature

Reply:
Actually it's looks like a [.filter on an array](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) here features are stored within the layer in Map<tileId, Map< batchId, C3DTFeature>> so if tilesC3DTFeatures is that object writing something like ([according stackoverflow](https://stackoverflow.com/questions/48707227/how-to-filter-a-javascript-map))

```
for ( const [tileId, tileC3DTFeatures] of tilesC3DTFeatures){
 const mapC3DTFeatureFiltered = new Map(
  [...tileC3DTFeatures]
  .filter(([batchId, c3DTFeature]) => //return tru when you want feature in the result )
 );
}
```

should do the work don't think we need an util for that, I didn't write it that way though (I just lean how to do it ^^)